manim/mobject/svg/svg_mobject.py

```diff
@@ -559,93 +562,140 @@ def generate_points(self):
             )
         )
         # Which mobject should new points be added to
-        self = self
+        prev_command = None
         for command, coord_string in pairs:
-            self.handle_command(command, coord_string)
+            self.handle_command(command, coord_string, prev_command)
+            prev_command = command.upper()
         # people treat y-coordinate differently
         self.rotate(np.pi, RIGHT, about_point=ORIGIN)
 
-    def handle_command(self, command, coord_string):
+    def handle_command(self, command, coord_string, prev_command):
         """Core logic for handling each of the various path commands."""
-        isLower = command.islower()
+        # Relative SVG commands are specified as lowercase letters
+        is_relative = command.islower()
         command = command.upper()
-        # new_points are the points that will be added to the curr_points
-        # list. This variable may get modified in the conditionals below.
-        points = self.points
-        new_points = self.string_to_points(coord_string)
 
-        if isLower and len(points) > 0:
-            new_points += points[-1]
+        # Keep track of the most recently completed point
+        start_point = (
+            self.points[-1] if len(self.points) > 0 else np.zeros((1, self.dim))
+        )
```

Comment:
```suggestion
        start_point = self.points[-1] if self.points else np.zeros((1, self.dim))
```

Reply:
I think the Black styling engine made this conversion for me; I'll check.

Reply:
I get a ValueError if I use `self.points` instead of `len(self.points) > 0` so I've undone this change.

Reply:
It's because self.points is not a list, but rather a numpy array. Maybe instead of `len(self.points) > 0` it should be `self.points.shape[0]`?

Reply:
If it is always guaranteed to be an array, then yes, please use `shape`